python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -824,33 +805,45 @@ def get_asset_check_key_for_test(
 
     # If the test is singular, but has multiple dependencies, infer the attached node from
     # from the dbt meta.
-    has_asset_key_config = bool(
+    attached_node_ref = (
         (
             test_resource_props.get("config", {}).get("meta", {})
             or test_resource_props.get("meta", {})
         )
         .get("dagster", {})
-        .get("asset_key", [])
-    )
-    attached_node_dbt_resource_props = (
-        (
-            manifest["nodes"].get(attached_node_unique_id)
-            or manifest["sources"].get(attached_node_unique_id)
-        )
-        if attached_node_unique_id
-        else test_resource_props
-    )
-    attached_node_asset_key = (
-        dagster_dbt_translator.get_asset_key(attached_node_dbt_resource_props)
-        if attached_node_unique_id or has_asset_key_config
-        else None
+        .get("ref", {})
     )
 
-    return (
-        AssetCheckKey(
-            name=test_resource_props["name"],
-            asset_key=attached_node_asset_key,
+    # Attempt to find the attached node from the ref.
+    if attached_node_ref:
+        ref_name, ref_package, ref_version = (
+            attached_node_ref["name"],
+            attached_node_ref.get("package"),
+            attached_node_ref.get("version"),
         )
-        if attached_node_asset_key
-        else None
+
+        project_name = manifest["metadata"]["project_name"]
+        if not ref_package:
```

Comment:
would be nice to have `package` (and less so `version`) under test but might be a pain